src/org/jitsi/impl/neomedia/transform/fec/AbstractFECReceiver.java

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright @ 2017 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.impl.neomedia.transform.fec;
+
+import org.jitsi.impl.neomedia.transform.*;
+import org.jitsi.service.configuration.*;
+import org.jitsi.service.libjitsi.*;
+import org.jitsi.service.neomedia.*;
+import org.jitsi.util.*;
+
+import java.util.*;
+
+/**
+ * A {@link PacketTransformer} which handles incoming fec packets.  This class
+ * contains only the generic fec handling logic.
+ *
+ * @author bgrozev
+ * @author bbaldino
+ */
+public abstract class AbstractFECReceiver
+    implements PacketTransformer
+{
+    /**
+     * The <tt>Logger</tt> used by the <tt>AbstractFECReceiver</tt> class and
+     * its instances to print debug information.
+     */
+    private static final Logger logger
+        = Logger.getLogger(AbstractFECReceiver.class);
+
+    class Statistics {
+        int numRxFecPackets;
+        int numRecoveredPackets;
+    }
+
+    /**
+     * Statistics for this fec receiver
+     */
+    protected Statistics statistics = new Statistics();
+
+    /**
+     * The SSRC of the fec stream
+     * NOTE that for ulpfec this might be the same as the associated media
+     * stream, whereas for flexfec it will be different
+     */
+    protected long ssrc;
+
+    /**
+     * Allow disabling of handling of ulpfec packets for testing purposes.
+     */
+    protected boolean handleFec = true;
```

Comment:
Instance fields go after all static fields/code.

Reply:
Feel free to ignore, since this comes from my old code which doesn't follow the convention...